src/main/java/seedu/address/model/task/DescriptionContainsKeywordPredicate.java

```diff
@@ -0,0 +1,31 @@
+package seedu.address.model.task;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Task}'s {@code Description} matches any of the keywords given.
+ */
+public class DescriptionContainsKeywordPredicate implements Predicate<Task> {
+    private final List<String> keywords;
+
+    public DescriptionContainsKeywordPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Task task) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(task.getDesc().value, keyword));
```

Comment:
Great to see that you allowed user to search tasks in a case-insensitive manner. 👍